src/Controller/Component/CrudComponent.php

```diff
@@ -649,7 +649,13 @@ protected function _loadListener($name)
                 throw new ListenerNotConfiguredException(sprintf('Listener "%s" is not configured', $name));
             }
 
+            // @todo: temporary hack to prevent `undefined index: className` error because `api` is in listeners list even though not enabled.
```

Comment:
Not odd at all, it was actually something very silly on my part again caused by an an old config setting for the ApiListener floating around in my code. Removing it solves all my problems. Undoing the hack right now.

```php
$this->Crud->config(['listeners.api.exceptionRenderer' => 'App\Error\ExceptionRenderer']);
```